os/20211014-s390x-secure-execution.md

```diff
@@ -0,0 +1,191 @@
+# IBM Z Secure Execution
+---
+
+# Theory of operation
+
+IBM Secure Execution for Linux® is a hardware-based security technology that is
+built into the IBM z15™ and LinuxONE III generation systems. It is designed to
+provide scalable isolation for individual workloads to help protect them from not
+only external attacks, but also insider threats. In traditional x86 ring architectures,
+the host can access the memory and data of guest applications freely, leading to the
+potential for malicious software to be proliferated throughout the entire system.
+Isolation between host and guest environments is necessary to help prevent system compromise.
+Users can enable Secure Execution feature on IBM zKVM Host and Guest VMs to isolate them from each other.
+
+That means:
+1. An encrypted Linux image (kernel + ramdisk + cmdline) is created using the host public key and a customer-specific key. Since the encryption keys are stored on the IBM Z hardware and firmware, the encrypted image can only be executed in a virtual machine on the host(s) it has been prepared for, and the image can’t be decrypted or tampered with outside of the designated host(s).
```

Comment:
Currently, the Machine Config Operator allows for configuring settings in the linux cmdline via MachineConfig. e.g.

```
apiVersion: machineconfiguration.openshift.io/v1
kind: MachineConfig
metadata:
  ...
spec:
  config:
    ignition:
      version: 3.1.0
  kernelArguments:
    - audit=1
```

Does this make it so that the kernel arguments are no longer configurable? or are there plans to make it so that changes to the cmdline through the MCO are re-encrypted with this mechanism?

Reply:
i don't know how MCO modifies kernel's cmdline, but if it's done via `rpm-ostree kargs`, than re-encryption would be done by `ostree`